src/utils/errorHandling.ts

```diff
@@ -0,0 +1,32 @@
+import type { Success, Failure, Result } from '@/shared.types';
+
+function createSuccessResult<T>(data?: T): Success<T> {
+  return { success: true, data };
+}
+
+function createFailureResult(error: string): Failure {
+  return { success: false, error };
+}
+
+function parseError(error: unknown): string {
```

Comment:
This function doesn't do any parsing, so perhaps a better name for it is something like `getErrorMessage`.